templates/netlify/remix.init/index.js

```diff
@@ -0,0 +1,108 @@
+const inquirer = require("inquirer");
+const fs = require("fs/promises");
+const { join } = require("path");
+const PackageJson = require("@npmcli/package-json");
+
+const { NETLIFY_EDGE_CI } = process.env;
+
+const filesToCopy = [
+  ["README.md"],
+  ["netlify.toml"],
+  ["server.js"],
+  ["remix.config.js"],
+  ["vscode.json", join(".vscode", "settings.json")],
+];
+
+const filesToModify = ["app/entry.server.tsx", "app/root.tsx"];
+
+async function modifyFilesForEdge(files, rootDirectory) {
+  const filePaths = files.map((file) => join(rootDirectory, file));
+  const contents = await Promise.all(
+    filePaths.map((path) => fs.readFile(path, "utf8"))
+  );
+
+  await Promise.all(
+    contents.map((content, index) => {
+      const newContent = content.replace(
+        /@remix-run\/node/g,
+        "@remix-run/deno"
+      );
+      return fs.writeFile(filePaths[index], newContent);
+    })
+  );
+}
+
+async function copyEdgeTemplateFiles(files, rootDirectory) {
+  for (const [file, target] of files) {
+    await fs.copyFile(
+      join(rootDirectory, "remix.init", file),
+      join(rootDirectory, target || file)
+    );
+  }
+}
+
+async function updatePackageJsonForEdge(directory) {
+  const packageJson = await PackageJson.load(directory);
+  const {
+    dependencies: {
+      "@remix-run/netlify": _netlify,
+      "@remix-run/node": _node,
+      ...dependencies
+    },
+    scripts,
+    ...restOfPackageJson
+  } = packageJson.content;
+
+  packageJson.update({
+    // dev script is the same as the start script for Netlify Edge, "cross-env NODE_ENV=production netlify dev"
+    scripts: { ...scripts, dev: scripts["start"] },
+    ...restOfPackageJson,
+    dependencies: { ...dependencies, "@remix-run/netlify-edge": "*" },
+  });
+
+  await packageJson.save();
+}
+
+async function main({ rootDirectory }) {
+  if (!(await shouldUseEdge())) {
+    return;
+  }
+
+  await Promise.all([
+    fs.mkdir(join(rootDirectory, ".vscode")),
+    copyEdgeTemplateFiles(filesToCopy, rootDirectory),
+  ]);
+
+  await Promise.all([
+    modifyFilesForEdge(filesToModify, rootDirectory),
+    updatePackageJsonForEdge(rootDirectory),
+  ]);
+}
+
+async function shouldUseEdge() {
+  if (typeof NETLIFY_EDGE_CI !== "undefined") {
+    // Skip the prompt if we're running in CI for Netlify Edge
+    return true;
+  }
+
```

Comment:
What do you think about always running the prompt, but select Edge when running CI (all in `scripts/deployment-test/netlify.mjs`) instead, like we do in our `remix-dev` tests with the `interactWithShell` helper function?

That way the `init` script doesn't need to know about the special CI case

```suggestion
```

Reply:
I agree it's not elegant that our remix.init script has to know about CI.

I didn't go with `interactShell` because of the way the deployment scripts run. From what I can tell, it pulls down a tarball from the repo and extracts the template directory and then installs packages and deploys. The remix.init script doesn't exist for other deploys like Cloudflare Workers, Deno, etc. so there is nothing in place to make it run which is why I import the `main()` from the Netlify Edge remix.init script and from what I can tell the Remix CLI does not run during these deploy scripts.

Reply:
We don't need Remix CLI to be run, we only need the prompt to be run & capture the process in order to be able to use `interactWithShell`

Reply:
I think supporting the env var `USE_NETLIFY_EDGE` (rather than `NETLIFY_EDGE_CI`) is a good thing, not just for these tests. It would allow headless install of templates in other scenarios too, and would be useful elsewhere.

I disagree with the idea of using `interactWithShell`. It makes sense for the dev tests, because those are testing the dev CLI itself. These are deployment tests though, so we shouldn't be trying to test the init script at the same time. We just happen to be using it to generate the template used for the deployment test.